geoh5py/shared/utils.py

```diff
@@ -833,3 +833,35 @@ def str2none(value):
     if value == "":
         return None
     return value
+
+
+def find_unique_name(name: str, names: list[str]) -> str:
+    """
+    Get a unique name not in a list of names.
+
+    :param name: The name to check.
+    :param names: The list of names to avoid.
+
+    :return: a unique name.
+    """
+
+    if name not in names:
+        return name
+
+    count = 1
+    while f"{name}({count})" in names:
+        count += 1
+
+    return f"{name}({count})"
+
+
+def remove_duplicates_in_list(input_list: list) -> list:
+    """
+    Remove duplicates from a list without changing the sorting.
+
+    :param input_list: the list to remove duplicates from.
+
+    :return: The sorted list
+    """
+    seen = set()
+    return [x for x in input_list if not (x in seen or seen.add(x))]
```

Comment:
```suggestion
    return sorted(set(input_list), key=input_list.index)
```

Reply:
I did not used list(sorted) because I wanted users to have the same order as their input data. 
If not, it's sorted by uids, which is kind of random.
Of course, this is a choice to make

Reply:
I am not following. The function I gave you above should do the same, but using C code instead of a python loop.